subsys/bluetooth/mesh/access.c

```diff
@@ -1614,20 +1741,39 @@ static int mod_set_pub(struct bt_mesh_model *mod, size_t len_rd,
 
 	err = bt_mesh_settings_set(read_cb, cb_arg, &pub, sizeof(pub));
 	if (err) {
-		LOG_ERR("Failed to set \'model-pub\'");
-		return err;
+		err = bt_mesh_settings_set(read_cb, cb_arg, &pub, sizeof(pub.base));
```

Comment:
Otherwise the customers will have to unprovision devices before upgrading to the new version, which is, for example, possible through P2P DFU. We already have this backward compatibility in, for example, CDB. Not that I like this, but this creates less pain for those who already is the stack.

Reply:
How about adding it under deprecation macro and then removing it after a couple of releases? 
My concern is that this solves the immediate problem and then becomes the useless ballast that we have to maintain.

Reply:
Added a deprecated option that disables the recovery code. By default, the option is disabled and the recovery code is present. This way it won't make a noise when building applications and won't require users to add anything in their project file. They can still remove support for the recovery code by enabling this option. After the deprecation period we will remove this option and the recovery code. Those who enabled this option will need to remove this after the deprecation period. Thanks for the suggestion!